R/readSource.R

```diff
@@ -118,7 +118,11 @@ readSource <- function(type,subtype=NULL,convert=TRUE) {
   if(source_missing) {
     # does a routine exist to download the source data?
     if(type %in% getSources(type="download")) {
-      downloadSource(type, subtype=subtype)
+      if (is.null(subtype)) {
+        downloadSource(type)
+      } else {
+        downloadSource(type, subtype=subtype)
+      }
```

Comment:
Is that distinction necessary? Isn't the default setting of subtype in downloadSource not NULL anyway?

Reply:
You're right, it's not necessary here.